spring-integration-core/src/main/java/org/springframework/integration/support/converter/ConfigurableCompositeMessageConverter.java

```diff
@@ -97,6 +99,7 @@ public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
 	@Override
 	public void afterPropertiesSet() {
 		if (this.registerDefaults) {
+			// TODO how to deal with this.beanFactory == null? just use DefaultConversionService.getSharedInstance()?
 			ConversionService conversionService = IntegrationUtils.getConversionService(this.beanFactory);
 			if (conversionService == null) {
 				conversionService = DefaultConversionService.getSharedInstance();
```

Comment:
If we consider `this.beanFactory` as `@Nullable` should we set `conversionService = DefaultConversionService.getSharedInstance();` as well or is some other behavior required?

Reply:
The `BeanFactoryAware.setBeanFactory(BeanFactory)` is not nullable.
Not sure why do yo u make such an assumption over here then...

Reply:
Well, the field `private BeanFactory beanFactory` is checked for `null` in several places throughout the whole framework although it is annotated with `@Nullable` only at very few places. I am fine with removing the annotation although it seems inconsistent regarding `@NonNullFields`

Reply:
With inconsistency I mean that  field `private BeanFactory beanFactory` as actually marked as `@NonNull` via `@NonNullFields` but actually can be `null` until `setBeanFactory` was invoked. But as I said, this holds for the major part of the framework.

Reply:
Actually, for `ConfigurableCompositeMessageConverter` I did not change anything. Currently, there is no check `this.beanFactory != null` in `main`.

I just removed the check from `DefaultDatatypeChannelMessageConverter.setBeanFactory()` as you told me, it must not be called with `null` as an argument. In this case, getting a NPE is the expected behavior, isn't it?

Reply:
That's correct, but since we talk about Spring, it is expected that these classes have to be registered as beans in the application context.
Therefore that `setBeanFactory()` is always called and really never with `null`.
So, we may assume that it is null by some end-user mistake or we can just bite a bullet and let it fly for time being!

Reply:
I am totally aware of that. `@NonNullFields` is more or less a tool for developers and actually no part of the public API. I am fine with every call you make.